### PR TITLE
Dockerized

### PR DESCRIPTION
**Pull request by harshkhandeparkar**
#### Changes
- Moved all source Python files to `src/`
- Created a `Dockerfile`

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+# Copy dependencies from the builder image
+COPY --from=builder /opt/venv /opt/venv
+ENV PATH="/opt/venv/bin:$PATH"
+
+# Copy source files
+WORKDIR /app
+COPY src/ .
+
+# Run the app
+CMD ["python", "main.py"]
```
Comment (rajivharlalka) [inline Dockerfile:30]: No Port exposed, would need one. 
Never run a prod app usingh direct python main.py. Use something like a uwgi gunicorn , take a look at wimp's dockerfile for help

**Review by rajivharlalka — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (amrav) [reply Dockerfile:30]: Hmm why should you not run a prod app using `python main.py`?

Comment (harshkhandeparkar) [reply Dockerfile:30]: The port is read from the environment variables. I thought it would be better to expose it through `docker run` instead.

Comment (rajivharlalka) [reply Dockerfile:30]: @amrav afaik gunicorn helps in managing python application by running multiple processes concurrently, which would improve the performance. I guess never is a wrong word but yes gunicorn should be preferred to run python applications atleast on production.

Comment (rajivharlalka) [reply Dockerfile:30]: @harshkhandeparkar we would still need it in the dockerfile and nevertheless would need a docker-compose too.

Comment (harshkhandeparkar) [reply Dockerfile:30]: > we would still need it in the dockerfile

How can I add a dynamic port in the Dockerfile?

Comment (harshkhandeparkar) [reply Dockerfile:30]: I can expose the default 8080 port though

Comment (rajivharlalka) [reply Dockerfile:30]: Didnt see that this is a tornado http server, seems the use of gunicorn wont be needed.

**Comment by proffapt** [comment]
Closing this as the new version has a complete re-write on `revamp` branch.
Please look into it (won't have that much changes in view of dockerization) and create a new PR.

**Pull request closed without merging**